lib/Core/Executor.cpp

```diff
@@ -276,38 +250,126 @@ namespace {
+            cl::cat(TerminationCat));
 
   cl::opt<bool>
   MaxMemoryInhibit("max-memory-inhibit",
-            cl::desc("Inhibit forking at memory cap (vs. random terminate) (default=on)"),
-            cl::init(true));
-
-  cl::opt<unsigned> RuntimeMaxStackFrames(
-      "max-stack-frames",
-      cl::desc("Terminates a state after the limit of stack frames is reached "
-               "(default=8192). Disable check with 0."),
-      cl::init(8192));
+                   cl::desc("Inhibit forking at memory cap (vs. random terminate) (default=on)"),
+                   cl::init(true),
+                   cl::cat(TerminationCat));
+
+  cl::opt<unsigned> RuntimeMaxStackFrames("max-stack-frames",
+                                          cl::desc("Terminate a state after this many stack frames (default=8192).  Disable check with 0."),
+                                          cl::init(8192),
+                                          cl::cat(TerminationCat));
+
+  cl::opt<std::string>
+  MaxInstructionTime("max-instruction-time",
+                     cl::desc("Allow a single instruction to take only this much time (default=0s (off)). Enables --use-forked-solver"),
+                     cl::cat(TerminationCat));
+
+  cl::opt<double>
+  MaxStaticForkPct("max-static-fork-pct",
+		   cl::init(1.),
+		   cl::desc("Maximum percentage spent by an instruction forking out of the forking of all instructions (default=1.0 (always))"),
+                   cl::cat(TerminationCat));
+
+  cl::opt<double>
+  MaxStaticSolvePct("max-static-solve-pct",
+		    cl::init(1.),
+		    cl::desc("Maximum percentage of solving time that can be spent by a single instruction over total solving time for all instructions (default=1.0 (always))"),
+                    cl::cat(TerminationCat));
+
+  cl::opt<double>
+  MaxStaticCPForkPct("max-static-cpfork-pct", 
+		     cl::init(1.),
+		     cl::desc("Maximum percentage spent by an instruction of a call path forking out of the forking of all instructions in the call path (default=1.0 (always))"),
+                     cl::cat(TerminationCat));
+
+  cl::opt<double>
+  MaxStaticCPSolvePct("max-static-cpsolve-pct",
+		      cl::init(1.),
+		      cl::desc("Maximum percentage of solving time that can be spent by a single instruction of a call path over total solving time for all instructions (default=1.0 (always))"),
+                      cl::cat(TerminationCat));
+
+
+
+  /*** Debugging options ***/
+
+  cl::OptionCategory DebugCat("Debugging options",
+                              "These are debugging options.");
+
+  /// The different query logging solvers that can switched on/off
+  enum PrintDebugInstructionsType {
+    STDERR_ALL, ///
+    STDERR_SRC,
+    STDERR_COMPACT,
+    FILE_ALL,    ///
+    FILE_SRC,    ///
+    FILE_COMPACT ///
+  };
+
+  llvm::cl::bits<PrintDebugInstructionsType> DebugPrintInstructions(
+      "debug-print-instructions",
+      llvm::cl::desc("Log instructions during execution."),
```

Comment:
Can `DebugCat` be added to this option as well?

Reply:
Good catch, missed that one due to some missing spacing (which I also fixed)